src/Simulation/Native/src/external/fusion.hpp

```diff
@@ -183,17 +227,17 @@ class Fusion{
 			for (auto idx : unhandled_ctrl){
 				new_ctrls.push_back(idx);
 				ctrl_set_.erase(idx);
-				set_.insert(idx);
+				target_set_.insert(idx);
 			}
 			for (auto &item : items_)
 				add_controls(item.get_matrix(), item.get_indices(), new_ctrls);
 		}
 	}
 	
-	IndexSet set_;
-	ItemVector items_;
-	IndexSet ctrl_set_;
-	Complex global_factor_;
+	mutable IndexSet target_set_; //set of qubits being acted on
```

Comment:
I still believe that it would be better to remove `const` from flush - it just makes no sense that *all* members of a class are mutable...